### ISSUE TITLE
Specify the key that's clashing when there's a key conflict

### ISSUE THREAD

**Issue by kendonB** [synkarius/Caster#49]
I believe I have a conflict between specs that's causing the below error. Unfortunately I don't know the source of the conflict and it would be good to have an error message that tells me what I did wrong. This particular error also brings the house down.
```
\_caster.py
Traceback (most recent call last):
  File "C:\NatLink\NatLink\MacroSystem\core\natlinkmain.py", line 331, in loadFile
    imp.load_module(modName,fndFile,fndName,fndDesc)
  File "C:\Users\kmbel\Documents\caster\_caster.py", line 33, in <module>
    control.init_nexus(_content_loader)
  File "C:\Users\kmbel\Documents\caster\castervoice\lib\control.py", line 7, in init_nexus
    _NEXUS = Nexus(content_loader)
  File "C:\Users\kmbel\Documents\caster\castervoice\lib\ctrl\nexus.py", line 83, in __init__
    self._grammar_manager.initialize()
  File "C:\Users\kmbel\Documents\caster\castervoice\lib\ctrl\mgr\grammar_manager.py", line 91, in initialize
    self._remerge_ccr_rules()
  File "C:\Users\kmbel\Documents\caster\castervoice\lib\ctrl\mgr\grammar_manager.py", line 192, in _remerge_ccr_rules
    ccr_rules_and_contexts = self._merger.merge_rules(active_ccr_mrs)
  File "C:\Users\kmbel\Documents\caster\castervoice\lib\merge\ccrmerging2\ccrmerger2.py", line 55, in merge_rules
    merged_rules = self._create_merged_rules(app_crs, non_app_crs)
  File "C:\Users\kmbel\Documents\caster\castervoice\lib\merge\ccrmerging2\ccrmerger2.py", line 98, in _create_merged_rules
    merged_non_app_crs_rule = self._merging_strategy.merge_into_single(non_app_crs)
  File "C:\Users\kmbel\Documents\caster\castervoice\lib\merge\ccrmerging2\merging\classic_merging_strategy.py", line 35, in merge_into_single
    incompat_index = indices_map[incompat_rcn]
KeyError: 'RStudioUserCCRRule'
```

The above was caused by me adding:
```
        SymbolSpecs.COMMENT:
            R(Key("cs-c"), rdescript="RStudio: Comment"),
```
to my rstudio rule when it already exists in the Rlang rule which was enabled. I will open another issue asking about better ways to deal with these sorts of conflicts without breaking the entire system.

**Comment by kendonB** [comment]
Possibly related to #16 ?

**Comment by synkarius** [comment]
This is definitely a bug, but it's irrelevant now that I've fixed #16. Note that I say irrelevant and not "fixed" because if someone turns the DetailsCompatibilityChecker back on in the future, this issue will come back, but it should be fine with the SimpleCompatibilityChecker.
